atomic_reactor/plugins/pre_fetch_sources.py

```diff
@@ -194,12 +211,44 @@ def check_lookaside_cache_usage(self):
         source_path = source.get()
         sources_cache_file = os.path.join(source_path, 'sources')
 
+        uses_cache = False
         if os.path.exists(sources_cache_file):
             if os.path.getsize(sources_cache_file) > 0:
-                raise RuntimeError('Repository is using lookaside cache, which is not allowed '
-                                   'for source container builds')
+                uses_cache = True
+
         source.remove_tmpdir()
 
+        if not uses_cache:
+            return
+
+        allowlist_cache_yaml = self._get_cache_allowlist()
+        should_raise = True
+
+        if allowlist_cache_yaml:
+            build_component = self.koji_build['package_name']
+            build_task = self.koji_build['extra'].get('container_koji_task_id')
+            task_info = self.session.getTaskInfo(build_task, request=True)
+            build_target = task_info['request'][1]
```

Comment:
Is it expected there might not be a 'container_koji_task_id' in `koji_build['extra']`? Is calling `getTaskInfo(None, ...)` what we want to do in that case?

If it's not expected, could we do `koji_build['extra']['container_koji_task_id']` instead?

Reply:
right it has to be there